src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/ListPackage/ListPackageCommandRunner.cs

```diff
@@ -139,26 +156,26 @@ public async Task ExecuteCommandAsync(ListPackageArgs listPackageArgs)
                                 if (printPackages)
                                 {
                                     var hasAutoReference = false;
-                                    ProjectPackagesPrintUtility.PrintPackages(packages, projectName, listPackageArgs, ref hasAutoReference);
+                                    ProjectPackagesPrintUtility.PrintPackages(packages, projectName, listPackageArgs, ref hasAutoReference, reportWriter);
                                     autoReferenceFound = autoReferenceFound || hasAutoReference;
                                 }
                             }
                         }
                     }
                     else
                     {
-                        Console.WriteLine(string.Format(Strings.ListPkg_ErrorReadingAssetsFile, assetsPath));
+                        reportWriter.WriteErrorReadingAssetsFile(assetsPath);
                     }
 
                     // Unload project
-                    ProjectCollection.GlobalProjectCollection.UnloadProject(project);
+                    projectCollection.UnloadProject(project);
```

Comment:
I know you're just refactoring the existing code, and I'm no MSBuild API expert, but I think that when loading MSBuild projects, you can tell it to load into a specific collection, rather than the global (static/shared) collection.

Now, this is largely irrelevant, since this project is a console app, not a library shared by many apps, and we know the process lifetime will be short, so nothing else will use MSBuild's global project collection in the process lifetime. But I think it makes sense to use a non-global project collection if we're just loading it, doing something quick, and then immediately unloading it.  I hate global/static state.